extra/onnx.py

```diff
@@ -24,6 +25,15 @@ def safe_numpy(t) -> np.ndarray:
     numpy_cache[t] = tmp
   return numpy_cache[t]
 
+# src: onnx/mapping.py
+# not supported: STRING = 8 COMPLEX64 = 14, COMPLEX128 = 15
+# NOTE: 17, 18, 19, 20 are float8
+DTYPE_MAP = {1:dtypes.float, 2:dtypes.uint8, 3:dtypes.int8, 4:dtypes.uint16, 5:dtypes.int16, 6:dtypes.int32, 7:dtypes.int64,
+              9:dtypes.bool, 10:dtypes.float16, 11:dtypes.double, 12:dtypes.uint32, 13:dtypes.uint64, 16:dtypes.bfloat16,
+              17:dtypes.float, 18:dtypes.float, 19:dtypes.float, 20:dtypes.float}
+# TODO: fix buffer_parse to use this and fix get_weight_and_biases to only use buffer_parse
+DTYPE_MAP = {i:dtype if Device.DEFAULT == "TORCH" or is_dtype_supported(dtype) else dtypes.int64 if dtypes.is_int(dtype) else dtypes.float32 for i,dtype in DTYPE_MAP.items()}
```

Comment:
hmm, `is_dtype_supported` is a test helper function and we probably don't want to rely on that for onnx support. what do you think?

Reply:
should work here if we use float for 10 (like the float8 ones)?

Reply:
Let me try.

Originally I wanted to use `is_dtype_supported` cuz it has very similar code to limiting dtypes as I had hard coded in `onnx.py`, but yeah importing from `test` is really weird.